servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConnectionAcceptorOffloadingTest.java

```diff
@@ -15,60 +15,187 @@
+                        offload ? ConnectExecutionStrategy.offloadAll() : ConnectExecutionStrategy.offloadNone()));
+        doRequest(builder);
+
+        assertThat("ConnectionAcceptor was not invoked", offloaded.get(), is(notNullValue()));
+        assertThat("Incorrect offloading for ConnectionAcceptor", offloaded.get(), is(offload));
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testSingleEarlyAcceptorOffloading(boolean offload) throws Exception {
+        AtomicReference<Boolean> offloaded = new AtomicReference<>();
+
+        HttpServerBuilder builder = HttpServers.forPort(0).appendEarlyConnectionAcceptor(new EarlyConnectionAcceptor() {
+            @Override
+            public Completable accept(final ConnectionInfo info) {
+                assertNotNull(info);
+                offloaded.set(!IoThreadFactory.IoThread.currentThreadIsIoThread());
+                return Completable.completed();
+            }
+
+            @Override
+            public ConnectExecutionStrategy requiredOffloads() {
+                return offload ? ConnectExecutionStrategy.offloadAll() : ConnectExecutionStrategy.offloadNone();
+            }
+        });
+        doRequest(builder);
 
-        try (ServerContext server = HttpServers.forPort(0)
-                .appendConnectionAcceptorFilter(factory)
-                .listenAndAwait(this::helloWorld)) {
+        assertThat("EarlyConnectionAcceptor was not invoked", offloaded.get(), is(notNullValue()));
+        assertThat("Incorrect offloading for EarlyConnectionAcceptor", offloaded.get(), is(offload));
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testSingleLateAcceptorOffloading(boolean offload) throws Exception {
+        AtomicReference<Boolean> offloaded = new AtomicReference<>();
+
+        HttpServerBuilder builder = HttpServers.forPort(0).appendLateConnectionAcceptor(new LateConnectionAcceptor() {
+            @Override
+            public Completable accept(final ConnectionInfo info) {
+                assertNotNull(info);
+                offloaded.set(!IoThreadFactory.IoThread.currentThreadIsIoThread());
+                return Completable.completed();
+            }
+
+            @Override
+            public ConnectExecutionStrategy requiredOffloads() {
+                return offload ? ConnectExecutionStrategy.offloadAll() : ConnectExecutionStrategy.offloadNone();
+            }
+        });
+        doRequest(builder);
+
+        assertThat("LateConnectionAcceptor was not invoked", offloaded.get(), is(notNullValue()));
+        assertThat("Incorrect offloading for LateConnectionAcceptor", offloaded.get(), is(offload));
+    }
+
+    /**
+     * Tests the offload merging and makes sure that if at least one is offloaded, both are.
+     */
+    @Test
+    void testMultipleEarlyAcceptorOffloading() throws Exception {
```

Comment:
Consider enhancing "multiple" tests by:
1. Adding the 3rd acceptor and verifying the order of acceptors execution.
2. Placing "offloaded" at 1, 2, 3rd positions to make sure that any offloaded acceptor enabled offloading for all of them.